packages/frontend/src/components/SchedulerModals/SchedulerModalBase/SchedulerForm.tsx

```diff
@@ -314,8 +320,8 @@ const SchedulerForm: FC<
                             renderRow={(key, index, remove) => {
                                 setShowDestinationLabel(false);
                                 const isSlack =
-                                    methods.getValues()?.targets?.[index]
-                                        ?.channel !== undefined;
+                                    'channel' in
+                                    methods.getValues()?.targets?.[index];
```

Comment:
if this is not absolutely necessary? if not I would just remove this change

Reply:
It is because the types were improved with the removal of the `methods` prop and introduction of better typed `methods` from the `useFormContext` hook. I've added a type guard that covers the scenario previously (to explicitly check that it's not `undefined`). what do you think @IrakliJani ? https://github.com/lightdash/lightdash/pull/5164/commits/63c3f25393f59437cc584148372cb69845ccaccb